R/unit.R

```diff
@@ -185,7 +187,13 @@ print.unit <- function(x, ...) {
 as.double.simpleUnit <- function(x, ...) as.double(unclass(x), ...)
 as.vector.simpleUnit <- function(x, ...) as.double(unclass(x), ...)
 
-is.unit <- function(x) inherits(x, 'unit')
+is.unit <- function(x) {
+    is_unit <- inherits(x, 'unit_v2')
+    if (!is_unit && inherits(x, 'unit')) {
```

Comment:
I'm not sure if this is a problem, but here we do ...
```
if !unit_v2 && unit
  error
```
... but in both as.unit() (R code) and asUnit() (C code), the pattern is ...
```
if unit
  if !unit_v2
    error
```
Is there a reason for the different pattern ?  Am I just being OCD ?

Reply:
The construct is just because `asUnit()` need to do additional stuff in the conditionals, whereas `is.unit` simply need to catch the specific condition where you pass in a unit that is not a unit_v2... The result is the same and the simplest form possible for both (I think)..

Don't know what you mean with `as.unit()` as it a direct call to `asUnit()`